src/session/session.c

```diff
@@ -838,10 +838,13 @@ user_has_valid_login_shell (const char **envp)
    * <pitti> https://xkcd.com/221/
    */
   const char *argv[] = { pwd->pw_shell, "-c", "exit 71;", NULL };
-  const int remap_fds[] = { -1, 2, -1 }; /* send stdout to stderr */
+  int devnull = open ("/dev/null", O_RDONLY);
```

Comment:
hate to do this to you, particularly if the bots are down, but this deserves at least an assert and maybe a proper ` if(...) abort();` or `return false;`

Reply:
I agree that this is a bit sloppy -- but if it fails, devnull will be -1, and thus fall back to the current behaviour. If /dev/null does not exist, there's little that we can do. But of course I'm happy to log a warning.

Reply:
But the premise of this patch is that the current behaviour is wrong...

If we can't open `/dev/null` then likely we have some fairly serious issues...

Reply:
I added an error check. I just err() out on failure, indeed without /dev/null a lot of things will go wrong anyway.